### ISSUE TITLE
Logins with potential duplicates should be able to be edited

### ISSUE THREAD

**Issue by eliserichards** [mozilla-mobile/fenix#12651]
Needs fix in app services: 

* https://github.com/mozilla-mobile/android-components/pull/7694
* https://github.com/mozilla/application-services/pull/3331

Currently, if a duplicate is found when a saved login is being edited, then changes are not able to be saved because of the way it's being handled in app services. This is manifesting in "false" saves, where the user is able to save an edit but their changes disappear. We need to 

1. get the app services latest version where this case is handled correctly and get that merged into AC
1. handle any new errors being thrown (if any) and make sure dupes are handled correctly when saving/showing errors to users.

### Steps to reproduce an error
**In `about:logins` on desktop:**

1. Create a new login: 
Website address: `https://www.google.com`
Username: `user1`
Password: `lskdflsdkfjs`
1. Create a second login with the same web address:
Website address: `https://www.google.com`
Username: `user2`
Password: `sdkjfhksd`
1. Sync to get logins on Fenix

**In Fenix:** 

1. Navigate to saved logins in the Settings menu
1. Open the newly created login and Edit the login
1. Change username or password
1. Save

**Comment by eliserichards** [comment]
Confirmed with Grisha that the fix is in the latest master for A-C (app service v61.0.10)

**Comment by eliserichards** [comment]
Waiting on https://github.com/mozilla-mobile/fenix/pull/12669

**Comment by ekager** [comment]
Looks like the blocking pieces have landed?
